frontend/packages/dev-console/src/components/pipelineruns/logs/MultiStreamLogs.tsx

```diff
@@ -45,6 +56,25 @@ export const MultiStreamLogs: React.FC<MultiStreamLogsProps> = ({ resource, task
       setAutoScroll(true);
     }
   }, [autoScroll, scrollDirection]);
+  const startDownloadAll = () => {
+    setDownloadAllStatus(true);
+    const buffer = new LineBuffer(null);
+    onDownloadAll()
+      .then(({ fileName, logs }: DownloadAll) => {
+        buffer.ingest(logs);
+        const blob = buffer.getBlob({
+          type: 'text/plain;charset=utf-8',
+        });
+        saveAs(blob, `${fileName}.log`);
```

Comment:
Why did you put this here? Move this back to the other file... I gave you a snippet, what was wrong with it?

You've made your utility no longer a download utility... it's a "prep for download" thus reducing its usefulness. Separation of concerns is **super** important in an application of this size.

Reply:
On error we would want to clear the buffer. that's why moved the buffer code into the callback itself. We are handling the error in the callback. So it did make sense to put that file download part in the callback as well

Reply:
Do we even care to clear on errors? It's done at this point anyways... Promise.all() triggering the .catch will be the last thing it does... the local scoped buffer is no longer reference-able and will be cleaned up by JS garbage collection.